src/picongpu/include/particles/bremsstrahlung/PhotonEmissionAngle.hpp

```diff
@@ -0,0 +1,264 @@
+        Probability(const float_64 gamma) : gamma2(gamma*gamma) {}
+
+        template<typename T_State>
+        void operator()(const T_State &p, T_State &dpdtheta, const float_64 theta) const
+        {
+            const float_64 theta2 = theta*theta;
+            const float_64 denom = 1.0 + gamma2 * theta2;
+
+            dpdtheta[0] = 3.0 * theta * gamma2 * (1.0 + gamma2*gamma2 * theta2*theta2) /
+                          (denom*denom*denom*denom);
+        }
+    };
+
+    /** Gives the absolute deviation of a delta, computed from a given theta, and a reference delta.
+     */
+    struct AimForDelta
+    {
+        const float_64 targetDelta;
+        const float_64 gamma;
+
+        /** constructor
+         *
+         * @param targetDelta reference delta
+         * @param gamma relativistic factor
+         */
+        AimForDelta(const float_64 targetDelta, const float_64 gamma) :
+            targetDelta(targetDelta), gamma(gamma) {}
+
+        float_64 delta(const float_64 theta, const float_64 gamma) const
+        {
+            namespace odeint = boost::numeric::odeint;
+
+            typedef boost::array<float_64, 1> state_type;
+
+            state_type integral_result = {0.0};
+            const float_64 lowerLimit = 0.0;
+            const float_64 upperLimit = theta;
+            const float_64 stepwidth = (upperLimit - lowerLimit) / 1000.0;
+            Probability integrand(gamma);
+            odeint::integrate(integrand, integral_result, lowerLimit, upperLimit, stepwidth);
+
+            return integral_result[0];
+        }
+
+        float_64 operator()(const float_64 theta) const
+        {
+            namespace pam = PMacc::algorithms::math;
+
+            return pam::abs(this->delta(theta, this->gamma) - this->targetDelta);
+        }
+    };
+
+    /** Return the maximal theta which corresponds to the maximal delta and a given gamma
+     *
+     * @param gamma relativistic factor
+     */
+    float_64 maxTheta(const float_64 gamma) const
+    {
+        AimForDelta aimForDelta(photon::MAX_DELTA, gamma);
+
+        std::pair<float_64, float_64> minimum;
+
+        minimum = boost::math::tools::brent_find_minima(
+            aimForDelta,
+            0.0,
+            M_PI,
+            std::numeric_limits<float_64>::digits);
+
+        return minimum.first;
+    }
+
+    /** computes the polar emission angle theta.
+     *
+     * @param delta uniformly distributed random number within [0, 1]
+     * @param gamma relativistic factor
+     * @param maxTheta maximal theta
+     */
+    float_64 theta(const float_64 delta, const float_64 gamma, const float_64 maxTheta) const
+    {
+        AimForDelta aimForDelta(delta, gamma);
+        const float_64 minTheta = 0.0;
+        std::pair<float_64, float_64> minimum;
+
+        minimum = boost::math::tools::brent_find_minima(
+            aimForDelta,
+            minTheta,
+            maxTheta,
+            std::numeric_limits<float_64>::digits);
```

Comment:
This is a quite strict limit - is this really required - or could or reduce this number and thus save computational time (or is computational time not an issue with this algorithm)?